tests/jobs/test_backfill_job.py

```diff
@@ -694,20 +712,21 @@ def test_backfill_run_rescheduled(self, dag_maker):
         ti.refresh_from_db()
         ti.set_state(State.UP_FOR_RESCHEDULE)
 
-        job = Job(executor=executor)
-        job_runner = BackfillJobRunner(
-            job=job,
-            dag=dag,
-            start_date=DEFAULT_DATE,
-            end_date=DEFAULT_DATE + datetime.timedelta(days=2),
-            rerun_failed_tasks=True,
-        )
-        run_job(job=job, execute_callable=job_runner._execute)
-        ti = TI(task=dag.get_task("test_backfill_run_rescheduled_task-1"), run_id=DEFAULT_DAG_RUN_ID)
-        ti.refresh_from_db()
-        assert ti.state == State.SUCCESS
+        for _ in self._mock_executor():
```

Comment:
Why the for loop?

Reply:
This is a neat bit of syntax that is using `_mock_executor()` as a generator since it cannot be used as a context manager (i.e. with  `with`) because the object that is yielded (the MockExecutor) does not implement `__aexit__` or `__aexit__`, you will get an exception if you try.

Reply:
🤯